src/weekly-test-summary/weekly-test-summary.service.ts

```diff
@@ -10,43 +12,48 @@ import { WeeklySystemIntegrityService } from '../weekly-system-integrity/weekly-
 export class WeeklyTestSummaryService {
 
   constructor(
+    private readonly dataSource: DataSource,
     private readonly map: WeeklyTestSummaryMap,
-    private readonly repository: WeeklyTestSummaryRepository,
     private readonly weeklySystemIntegrityService: WeeklySystemIntegrityService,
   ) {}
+
   async getWeeklyTestSummariesByLocationIds(
     monitoringLocationIds: string[],
     params: EmissionsParamsDTO,
   ): Promise<WeeklyTestSummaryDTO[]> {
-    const results = await this.repository.export(
-      monitoringLocationIds,
-      params.year,
-      params.quarter,
-    );
-
-    return this.map.many(results);
+    return withSlaveConnection(this.dataSource, async (replicaManager: EntityManager) => {
+      const weeklyTestSummaryRepository = new WeeklyTestSummaryRepository(replicaManager);
+      const results = await weeklyTestSummaryRepository.export(
+        monitoringLocationIds,
+        params.year,
+        params.quarter,
+      );
+      return this.map.many(results);
+    });
   }
 
   async export(
     monitoringLocationIds: string[],
     params: EmissionsParamsDTO,
   ): Promise<WeeklyTestSummaryDTO[]> {
-    const weeklyTestSummaries = await this.getWeeklyTestSummariesByLocationIds(
-      monitoringLocationIds,
-      params,
-    );
-
-    if (weeklyTestSummaries) {
-      const weeklySystemIntegrityData = await this.weeklySystemIntegrityService.export(
-        weeklyTestSummaries?.map(i => i.id),
+    return withSlaveConnection(this.dataSource, async () => {
```

Comment:
It looks like this call is unnecessary, since the replica entity manager isn't used.
```suggestion
```